main.go

```diff
@@ -145,3 +185,76 @@ func loadConfig(cfgJSON *extapi.JSON) (customDNSProviderConfig, error) {
 
 	return cfg, nil
 }
+
+func (c *ns1DNSProviderSolver) setNS1Client(ch *v1alpha1.ChallengeRequest, cfg ns1DNSProviderConfig) error {
+	ref := cfg.APIKeySecretRef
+	if ref.Key == "" {
+		return fmt.Errorf(
+			"no key '%s' in secret '%s/%s'",
+			ref.Key,
```

Comment:
Is this error message accurate?  This seems to be checking that a key was given, not for it's presence in the secret.

Reply:
i think this one is accurate, it would read `no key 'apiKey' in secret cert-manager/ns1-credentials`

Reply:
but it's only returned if `ref.Key == ""` so wouldn't it read `no key '' in secret cert-manager/ns1-credentials`?

Reply:
/me hand hits forehead